sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -78,26 +87,46 @@
    */
   public static <T, IdT> WithRepresentativeValues<T, IdT> withRepresentativeValueFn(
       SerializableFunction<T, IdT> fn) {
-    return new WithRepresentativeValues<T, IdT>(fn, null);
+    return new WithRepresentativeValues<>(fn, null);
+  }
+
+  private static <T> void validateWindowFn(PCollection<T> in) {
```

Comment:
nits: (1) technically, you are validating the whole `WindowingStrategy`, not the `WindowFn` and (2) this should just take the `WindowingStrategy` not the whole `PCollection`.